src/ivoc/ivocvect.cpp

```diff
@@ -165,17 +166,17 @@ extern void notify_freed_val_array(double*, size_t);
 
 extern int hoc_return_type_code;
 
-IvocVect::IvocVect(Object* o) : ParentVect(){obj_ = o; label_ = NULL; MUTCONSTRUCT(0)}
-IvocVect::IvocVect(int l, Object* o) : ParentVect(l){obj_ = o; label_ = NULL; MUTCONSTRUCT(0)}
-IvocVect::IvocVect(int l, double fill_value, Object* o) : ParentVect(l, fill_value){obj_ = o; label_ = NULL; MUTCONSTRUCT(0)}
-IvocVect::IvocVect(IvocVect& v, Object* o) : ParentVect(v) {obj_ = o; label_ = NULL; MUTCONSTRUCT(0)}
+IvocVect::IvocVect(Object* o) {obj_ = o; label_ = NULL; MUTCONSTRUCT(0)}
+IvocVect::IvocVect(int l, Object* o) : vec_(l) {obj_ = o; label_ = NULL; MUTCONSTRUCT(0)}
+IvocVect::IvocVect(int l, double fill_value, Object* o) : vec_(l, fill_value){obj_ = o; label_ = NULL; MUTCONSTRUCT(0)}
+IvocVect::IvocVect(IvocVect& v, Object* o) : vec_(v.vec_) {obj_ = o; label_ = NULL; MUTCONSTRUCT(0)}
 
 IvocVect::~IvocVect(){
 	MUTDESTRUCT
 	if (label_) {
 		delete [] label_;
 	}
-	notify_freed_val_array(vec(), capacity());
+	notify_freed_val_array(vec_.data(), vec_.capacity());
```

Comment:
In many places you replaced capacity() by size().  I realize it is necessary for legacy API to implement capacity()

Reply:
Well `capacity` actually represents `size` in `doubleAVec->doubleVec`: 
```cpp
inline int doubleVec::capacity()
{
  return len;
}
```

`vector_capacity` is used in a lot of models, where we can see it's used as `size`:
```bash
Binary file ./247968/mod/x86_64/libnrnmech.dylib matches
./247968/mod/x86_64/vecevent.c:255:			size = vector_capacity(vv);
./247968/mod/x86_64/vecevent.c:441:  "			size = vector_capacity(vv);\n"
Binary file ./247968/mod/x86_64/.libs/libnrnmech.so matches
Binary file ./247968/mod/x86_64/vecevent.o matches
./222359/mod/vecevent.mod:44:			size = vector_capacity(vv);
./222359/mod/x86_64/vecevent.mod:44:			size = vector_capacity(vv);
Binary file ./222359/mod/x86_64/libnrnmech.dylib matches
./222359/mod/x86_64/vecevent.c:250:			size = vector_capacity(vv);
./222359/mod/x86_64/vecevent.c:391:  "			size = vector_capacity(vv);\n"
Binary file ./222359/mod/x86_64/.libs/libnrnmech.so matches
Binary file ./222359/mod/x86_64/vecevent.o matches
./262670/mod/vecevent.mod:49:			size = vector_capacity(vv);
./262670/mod/x86_64/vecevent.mod:49:			size = vector_capacity(vv);
Binary file ./262670/mod/x86_64/libnrnmech.dylib matches
./262670/mod/x86_64/vecevent.c:250:			size = vector_capacity(vv);
./262670/mod/x86_64/vecevent.c:416:  "			size = vector_capacity(vv);\n"
Binary file ./262670/mod/x86_64/.libs/libnrnmech.so matches
Binary file ./262670/mod/x86_64/vecevent.o matches
./185858/mod/vecst.mod:2123:  sz = vector_capacity(obv->u.this_pointer);
``` 

> I realize it is necessary for legacy API to implement capacity()

That's true for internal API, public API is providing `size` https://neuronsimulator.github.io/nrn/hoc/programming/math/vector.html?highlight=vector#Vector.size 
I guess it would make more sense to see `capacity` as a misnomer 
```cpp
extern int vector_capacity(IvocVect*); //ivocvect.h conflicts with STL
```

Reply:
>  see capacity as a misnomer
I agree!